uiautomationutilities/perf_tool/readme.md

```diff
@@ -1,155 +1,157 @@
+- [Email report](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/paul/update-perf-testing-documentation/uiautomationutilities/perf_tool/readme.md#email-report)
+
+### Definition of terms:
+[**PerfMarker / CodeMarker:**](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/PerfConstants.java#L36) A marker in code that results in an event to capture timestamp and other details as needed (Memory, Thread ID, CorrelationID etc. and write to file). The data generated from a CodeMarker can be used for Perf, logging, Telemetry etc.
+
+[**Measurement:**](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/uiautomationutilities/perf_tool/PerfDataConfiguration.xml#L4) A pair of markers identified for a transaction. E.g. If can have marker M1 at the beginning of a silent request and another when we successfully received a token say M2. Now M1 and M2 can be paired together if we were interested in measuring time between these. Pairing is an outside concept and has no impact from a code perspective.
+
+[**Scenario:**](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/uiautomationutilities/perf_tool/PerfDataConfiguration.xml#L18-L25) An E2E scenario that we would be interested in measuring. E.g. Silent Token refresh on a 4G network when requesting a token for graph. A scenario can be associated with one or more measurements.
+
+**Base build:** This is a build against which we compare performance of future builds. It would typically be the last release. It is configured on the Perf pipeline using the `baseBuildID` variable
+
+**Basefiles:** These are the code marker files during the run of the base build and help us compare the current run against the base run
+
+**Targetfiles:** These are the code marker files generated during the current run
+
+
+### Steps to add a new marker and see it in the report:
+- If a new scenario has to be created, come up with a scenario code comprised of 3 digits and add it to the codemarker definitions in file [PerfConstants.java](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/PerfConstants.java), for example some of the existing scenario codes are 100 and 200.
+- In the common codebase, and add codemarkers (e.g. start of the activity to be measured and the end of the activity to be measured) at desired places. See example in [CommandDispatcher.java](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java#L197). Example events lines are given as follows:
+	- `CodeMarkerManager.getInstance().markCode(CodeMarkersConstants.ACQUIRE_TOKEN_SILENT_START);`
```

Comment:
Is there a convention to follow when adding new `CodeMarkerConstants`? Are are the strings/ints "random"?

Reply:
Addressed, see https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/paul/update-perf-testing-documentation/uiautomationutilities/perf_tool/readme.md#changes-needed-in-common---javakotlin-code

Reply:
I would be curious to know the depth of the any change required to abandon the numeric backing of these strings.  Because this seems like a completely unnecessary extra step.  Not that this matters for these instructions, but that's where we are.